### DISCUSSION TITLE
Prune old database notifications

### DISCUSSION THREAD

**Discussion by smknstd** [laravel/framework#40124] [Ideas] [answered]
It would be great if I was able to [prune](https://laravel.com/docs/8.x/eloquent#pruning-models) old [database notifications](https://laravel.com/docs/8.x/notifications#database-notifications) without extending notifications with custom model.

What do you think ?

**Comment by yilanboy** [comment, marked as answer]
I create a `Notification` model and use [Pruning Models](https://laravel.com/docs/8.x/eloquent#pruning-models) to clean the stale notification records.

```php
<?php

namespace App\Models;

use Illuminate\Database\Eloquent\Builder;
use Illuminate\Database\Eloquent\Factories\HasFactory;
use Illuminate\Database\Eloquent\Model;
use Illuminate\Database\Eloquent\MassPrunable;

class Notification extends Model
{
    use HasFactory, MassPrunable;

    public function prunable(): Builder
    {
        return static::whereNotNull('read_at')
            ->where('read_at', '<=', now()->subWeek());
    }
}
```
then add this prune task in `Console/Kernel.php`
```php
protected function schedule(Schedule $schedule)
{
    $schedule->command('model:prune')->daily();
}
```
It's worked for me.

**Comment by smknstd** [comment]
Thanks  @YilanBoy, I just wrote a really simple blog post, I hope it can help someone someday: https://medium.com/code16/automatically-delete-read-database-notifications-in-laravel-59871ee3c98
